master/internal/core.go

```diff
@@ -484,23 +484,37 @@ func (m *Master) Run() error {
 	})
```

Comment:
non-blocking: Could you change the 301 in the line above this to `http.StatusMovedPermanently`? Slightly off-topic, perhaps, but I don't feel like making a PR just for that.